### DISCUSSION TITLE
fetch vs node-fetch in Next.js api

### DISCUSSION THREAD

**Discussion by jeffyamada** [Xunnamius/next-test-api-route-handler#820] [Q&A] [answered]
Hi all! 

Ok, I think this might be a silly question but I finally have an API all testing properly BUT I've had to import node-fetch in my api, when I don't need to do this when simply running the api in Next.js + Vercel. I'm not sure why Next.js/Vercel don't seem to mind simply using `fetch` without  importing `node-fetch`. I presume Next.js is using node 18+ where this is included, but then can't figure out why when I run my tests it's not included when I'm running 18+.

Simply put: I can run my api without `node-fetch` normally, but when I run tests I need to use `import fetch from 'node-fetch'`.

Any help or input would be greatly appreciated!

**Comment by Xunnamius** [comment, marked as answer]
This does indeed sound like something to do with later Node versions with built-in fetch vs older Node versions that rely on third-party fetch. It could be the case that the `node-fetch` package does something that disables the built-in fetch, or that Node disables built-in fetch when it detects one of the third party fetch libraries, but I haven't seen any documentation that describes such behavior from Node or the `node-fetch` package.

If you make a minimal reproducible example I can take a look.
